server/graphql_schemas/event/schema.py

```diff
@@ -357,12 +359,46 @@ class SlackChannelsList(graphene.ObjectType):
     class Meta:
         interfaces = (relay.Node,)
 
+
+class ShareEvent(relay.ClientIDMutation):
+    class Input:
+        event_id = graphene.ID()
+        channel_id = graphene.ID()
+
+    event = graphene.Field(EventNode)
+
+    @classmethod
+    def mutate_and_get_payload(cls, root, info, **input):
+        event_id = input.get('event_id')
+        channel_id = input.get('channel_id')
+        event_url = f"{dotenv.get('FRONTEND_BASE_URL')}/{event_id}"
+        event = Event.objects.get(pk=event_id)
+
+        try:
+            start_date = parse(event.start_date)
+            message = (f"*A new event has been created by <@{event.creator.slack_id}>.*\n"
+                       f"> *Title:* {event.title}\n"
+                       f"> *Description:* {event.description}\n"
+                       f"> *Venue:* {event.venue}\n"
+                       f"> *Date:*  {start_date.strftime('%d %B %Y, %A')} \n"
+                       f"> *Time:*  {start_date.strftime('%H:%M')}")
+            blocks = new_event_message(
+                message, event_url, event_id, event.featured_image)
+
+            notify_channel(
+                blocks, "New upcoming event from Andela socials", channel_id)
+
+        except ValueError as e:
+            raise GraphQLError("An Error occurred. \n{}".format(e))
```

Comment:
I don't think we ideally want to return raw GraphQL error object messages to the client. We could have a more precise custom message here.